CMakeLists.txt

```diff
@@ -1,35 +1,126 @@
-cmake_minimum_required(VERSION 3.17...3.22)
+cmake_minimum_required(VERSION 3.17...3.26)
```

Comment:
Is the upper bound necessary here? It doesn't seem like cmake complains about this, so not quite sure what it's for..